x-pack/plugins/security_solution/public/management/pages/policy/view/tabs/policy_tabs.tsx

```diff
@@ -85,6 +86,25 @@ export const PolicyTabs = React.memo(() => {
   const isInProtectionUpdatesTab = usePolicyDetailsSelector(isOnProtectionUpdatesView);
   const policyId = usePolicyDetailsSelector(policyIdFromParams);
 
+  const [showUnsavedChangesModalForTab, setShowUnsavedChangesModalForTab] =
```

Comment:
for me, the name of this state doesn't indicate that it contains the tab the user wants to *switch to*, but tbh I dont have a better name idea, as it is doing 2 things at once: indicating whether we need to display the modal, and where to switch when the user clicks Continue.

still, I think somehow we should modify this, because it is quite implicit

I can suggest either to use 2 states (or an object as one combined state) for the 2 different goals, or we could store the modal istelf in a `modal` state, enclosing the information about which tab was selected (it could be created in `onTabClickHandler()`)

Reply:
Yup, I sat there for a while thinking about how to do it best. Anyway, did some changes [here](https://github.com/elastic/kibana/pull/176605/commits/5aa8cb320e1b1d87c5c1f4a68d86b66372732d70), let me know what you think.

Reply:
that looks really nice and explicit, thanks!